sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java

```diff
@@ -62,20 +64,22 @@ public void testWriteRead() throws Exception {
       KV<String, String> kv = KV.of("key " + i, "value " + i);
       data.add(kv);
     }
-    PCollection<KV<String, String>> write = pipeline.apply(Create.of(data));
+    PCollection<KV<String, String>> write = writePipeline.apply(Create.of(data));
     write.apply(RedisIO.write().withConnectionConfiguration(connection));
 
-    pipeline.run();
+    writePipeline.run();
 
-    PCollection<KV<String, String>> read = pipeline.apply("Read",
+    PCollection<KV<String, String>> read = readPipeline.apply("Read",
         RedisIO.read().withConnectionConfiguration(connection).withKeyPattern("key*"));
     PAssert.that(read).containsInAnyOrder(data);
 
-    PCollection<KV<String,  String>> readNotMatch = pipeline.apply("ReadNotMatch",
+    readPipeline.run();
```

Comment:
These two (read and readNoKey) could have been one pipeline :) As long as you don't add new transforms after you've called .run(), you're fine.